src/channel/ws/ws_0_8.ts

```diff
@@ -0,0 +1,443 @@
+    });
+    if (status) {
+      this.onUnsubscribeLocal(subscriptionId);
+      this.onUnsubscribe(subscriptionId);
+    }
+    return status;
+  }
+
+  /**
+   * await while subscription is unsubscribed
+   * @param forSubscriptionId if defined trigger on subscriptionId else trigger on any
+   * @returns subscriptionId | onerror(Event)
+   * @example
+   * ```typescript
+   * const subscriptionId = await webSocketChannel.waitForUnsubscription();
+   * ```
+   */
+  public async waitForUnsubscription(forSubscriptionId?: SUBSCRIPTION_ID) {
+    // Wait for unsubscription event
+    return new Promise((resolve, reject) => {
+      if (!this.websocket || this.websocket.readyState !== WebSocket.OPEN) {
+        reject(new Error('WebSocket not available or not open. Cannot wait for unsubscription.'));
+        return;
+      }
+
+      let localOnError: (event: Event) => void;
+
+      const localOnUnsubscribe = (subscriptionId: SUBSCRIPTION_ID) => {
+        if (this.onUnsubscribeLocal === localOnUnsubscribe) {
+          this.websocket.removeEventListener('error', localOnError);
+
+          if (forSubscriptionId === undefined) {
+            resolve(subscriptionId);
+          } else if (subscriptionId === forSubscriptionId) {
+            resolve(subscriptionId);
+          }
+        }
+      };
+
+      localOnError = (event: Event) => {
+        if (this.onUnsubscribeLocal === localOnUnsubscribe) {
+          this.websocket.removeEventListener('error', localOnError);
+          reject(
+            new Error(
+              `WebSocket error while waiting for unsubscription of ${forSubscriptionId || 'any subscription'}: ${event.type || 'Unknown error'}`
+            )
+          );
+        }
+      };
+
+      this.onUnsubscribeLocal = localOnUnsubscribe;
+      this.websocket.addEventListener('error', localOnError);
+    });
+  }
+
+  /**
+   * Reconnect re-create this.websocket instance
+   */
+  public reconnect() {
+    this.websocket = new WebSocket(this.nodeUrl);
+
+    this.websocket.addEventListener('open', this.onOpen.bind(this));
+    this.websocket.addEventListener('close', this.onCloseProxy.bind(this));
+    this.websocket.addEventListener('message', this.onMessageProxy.bind(this));
+    this.websocket.addEventListener('error', this.onError.bind(this));
+  }
+
+  private onCloseProxy(ev: CloseEvent) {
+    this.websocket.removeEventListener('open', this.onOpen);
+    this.websocket.removeEventListener('close', this.onCloseProxy);
+    this.websocket.removeEventListener('message', this.onMessageProxy);
+    this.websocket.removeEventListener('error', this.onError);
+    this.onClose(ev);
+  }
+
+  private onMessageProxy(event: MessageEvent<any>) {
+    const message: WebSocketEvent = JSON.parse(event.data);
+
+    // Check if it's a subscription event
+    if (
+      message.method &&
+      'params' in message &&
+      message.params &&
+      typeof message.params === 'object' &&
+      'subscription_id' in message.params
```

Comment:
Looks like it could be shortened to `message.method && isObject(message.params) && 'subscription_id' in message.params`.

Reply:
https://github.com/starknet-io/starknet.js/pull/1416/commits/aea7f527eb4a881bc5172e511c96abd2affd36d5